### PR TITLE
Adding irods publisher

### PR DESCRIPTION
**Pull request by ke4**

### CODE REVIEW

File: src/perl/lib/wtsi_clarity/irods/irods_publisher.pm [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+=head1 NAME
+
+wtsi_clarity::irods::irods_publisher
+
+=head1 SYNOPSIS
+
+  wtsi_clarity::irods::irods_publisher->new(
+       process_url => 'http://clarity-ap:8080/processes/3345',
+       step_url    => 'http://testserver.com:1234/here/steps/24-98970',
+  )->run();
```
Comment (jbeast) [inline src/perl/lib/wtsi_clarity/irods/irods_publisher.pm]: Is this right?

File: src/perl/lib/wtsi_clarity/irods/irods_publisher.pm [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+=back
+
+=head1 AUTHOR
+
+Karoly Erdos E<lt>ke4@sanger.ac.ukE<gt>
+
+=head1 LICENSE AND COPYRIGHT
+
+Copyright (C) 2014 Genome Research Ltd.
```
Comment (jbeast) [inline src/perl/lib/wtsi_clarity/irods/irods_publisher.pm]: 2015

File: src/perl/lib/wtsi_clarity/epp/generic/manifest.pm [hunk 3]
```diff
@@ -62,6 +74,25 @@ has 'container_id' => (
   required  => 0,
 );
 
+has 'publish_to_irods' => (
```
Comment (jbeast) [inline src/perl/lib/wtsi_clarity/epp/generic/manifest.pm]: Does this default to true or false?

File: src/perl/lib/wtsi_clarity/irods/irods_publisher.pm (hunk 1 above)
Comment (ke4) [inline src/perl/lib/wtsi_clarity/irods/irods_publisher.pm]: Oh, no.
I will correct it.

File: src/perl/lib/wtsi_clarity/irods/irods_publisher.pm [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+=over
+
+=item Moose
+
+=item Carp
+
+=item Readonly
+
+=item XML::LibXML
```
Comment (jbeast) [inline src/perl/lib/wtsi_clarity/irods/irods_publisher.pm]: Don't think this is a dependency

File: src/perl/lib/wtsi_clarity/epp/generic/manifest.pm (hunk 3 above)
Comment (ke4) [inline src/perl/lib/wtsi_clarity/epp/generic/manifest.pm]: false, I will add a default property...

**Pull request merged**